packages/animations/test/page_transition_switcher_test.dart

```diff
@@ -0,0 +1,622 @@
+      (WidgetTester tester) async {
+    final UniqueKey containerOne = UniqueKey();
+    final UniqueKey containerTwo = UniqueKey();
+    final UniqueKey containerThree = UniqueKey();
+
+    // Insert three unique children so that we have some previous children.
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: PageTransitionSwitcher(
+          duration: const Duration(milliseconds: 100),
+          child: Container(key: containerOne, color: const Color(0xFFFF0000)),
+          transitionBuilder: _transitionBuilder,
+        ),
+      ),
+    );
+
+    await tester.pump(const Duration(milliseconds: 10));
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: PageTransitionSwitcher(
+          duration: const Duration(milliseconds: 100),
+          child: Container(key: containerTwo, color: const Color(0xFF00FF00)),
+          transitionBuilder: _transitionBuilder,
+        ),
+      ),
+    );
+
+    await tester.pump(const Duration(milliseconds: 10));
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: PageTransitionSwitcher(
+          duration: const Duration(milliseconds: 100),
+          child: Container(key: containerThree, color: const Color(0xFF0000FF)),
+          transitionBuilder: _transitionBuilder,
+        ),
+      ),
+    );
+
+    await tester.pump(const Duration(milliseconds: 10));
+
+    expect(find.byType(FadeTransition), findsNWidgets(3));
+    expect(find.byType(ScaleTransition), findsNWidgets(3));
+    expect(find.byType(SlideTransition), findsNothing);
+    expect(find.byType(SizeTransition), findsNothing);
+
+    Widget newTransitionBuilder(
+        Widget child, Animation<double> primary, Animation<double> secondary) {
+      return SlideTransition(
+        position: Tween<Offset>(begin: Offset.zero, end: const Offset(20, 30))
+            .animate(primary),
+        child: SizeTransition(
+          sizeFactor: Tween<double>(begin: 10, end: 0.0).animate(secondary),
+          child: child,
+        ),
+      );
+    }
+
+    // Now set a new transition builder and make sure all the previous
+    // transitions are replaced.
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: PageTransitionSwitcher(
+          duration: const Duration(milliseconds: 100),
+          child: Container(key: containerThree, color: const Color(0x00000000)),
+          transitionBuilder: newTransitionBuilder,
+        ),
+      ),
+    );
+
+    await tester.pump(const Duration(milliseconds: 10));
+
+    expect(find.byType(FadeTransition), findsNothing);
+    expect(find.byType(ScaleTransition), findsNothing);
+    expect(find.byType(SlideTransition), findsNWidgets(3));
+    expect(find.byType(SizeTransition), findsNWidgets(3));
+  });
+}
+
+class StatefulTest extends StatefulWidget {
```

Comment:
Perhaps rename to make it clearer:
```suggestion
class StatefulTestWidget extends StatefulWidget {
```